src/main/java/com/linkedin/kmf/consumer/NewConsumer.java

```diff
@@ -15,33 +15,99 @@
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.clients.consumer.OffsetCommitCallback;
 import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.internals.Topic;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/*
- * Wrap around the new consumer from Apache Kafka and implement the #KMBaseConsumer interface
+
+/**
+ * Wraps around the new consumer from Apache Kafka and implements the #KMBaseConsumer interface
  */
 public class NewConsumer implements KMBaseConsumer {
 
   private final KafkaConsumer<String, String> _consumer;
   private Iterator<ConsumerRecord<String, String>> _recordIter;
-  private static final Logger LOG = LoggerFactory.getLogger(NewConsumer.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(NewConsumer.class);
   private static long lastCommitted;
+  protected String _targetConsumerGroupId;
+  protected String _consumerGroupPrefix = "__shadow_consumer_group-";
+  protected int _consumerGroupSuffixCandidate = 0;
```

Comment:
Do these need to be data members a not local variables?   Why is_consumerGroupPrefix mutable?